backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,283 @@
+    period: 'all',
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
+    workspaceIds: []
+  };
+
+  return (got
+    .post(resource, {
+      json: modifier,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + process.env.LG_API_KEY
+      }
+    })
+    .json() as unknown) as LGThreatResponse[];
+}
+
+async function saveAndPullDomains(response, organizationId, scanId) {
+  const domains: Domain[] = [];
+  const data = response;
+  for (const l of data.results) {
+    if (validateIPAddress(l.left.name)) {
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: l.left.name,
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: true,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    } else {
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: l.left.name,
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: false,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    }
+  }
+  await saveDomainsToDb(domains);
+
+  await connectToDatabase();
+
+  let pulledDomains = Domain.createQueryBuilder('domain')
+    .leftJoinAndSelect('domain.organization', 'organization')
+    .andWhere('ip IS NOT NULL');
+
+  pulledDomains = pulledDomains.andWhere('domain.organization=:org', {
+    org: organizationId
+  });
+
+  return pulledDomains.getMany();
+}
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName, scanId } = commandOptions;
+
+  console.log('Running lookingGlass on organization', organizationName);
+
+  const collections: LGCollectionsResponse[] = await getCollectionForCurrentWorkspace();
+  const ipsAndDomains: string[] = [];
+  const vulnerabilities: Vulnerability[] = [];
+
+  for (const line of collections) {
+    // Match the organization to the LookingGlass Collection
+    if (organizationName === line.name) {
+      const collectionID = line.id;
+      console.log(line.name);
+      // Query LookingGlass for the Threat info
+      const data: LGThreatResponse[] = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId,
+        scanId
+      );
+      const d = new Date();
+      d.setMonth(d.getMonth() - 1);
+      console.log(d);
+      for (const l of data['results']) {
+        // Create a dictionary of relevant fields from the API request
+        const lastSeen = new Date(l.firstSeen);
+        if (lastSeen > d) {
+          const val = {
+            firstSeen: new Date(l.firstSeen),
```

Comment:
We should define a type (interface) for this data structure we're creating here (because it's custom and we access things from it elsewhere).